include/parallel/parallel_implementation.h

```diff
@@ -3024,6 +3024,52 @@ inline void Communicator::allgather(T sendval,
 
 
 
+template <>
+inline void Communicator::allgather(std::string sendval,
+                                    std::vector<std::string> & recv) const
+{
+  LOG_SCOPE ("allgather()","Parallel");
+
+  libmesh_assert(this->size());
+  recv.assign(this->size(), "");
+
+  std::vector<int>
+    sendlengths  (this->size(), 0),
+    displacements(this->size(), 0);
+
+  // first comm step to determine buffer sizes from all processors
+  const int mysize = static_cast<int>(sendval.size());
+  this->allgather(mysize, sendlengths);
+
+  // Find the total size of the final array and
+  // set up the displacement offsets for each processor
+  unsigned int globalsize = 0;
+  for (unsigned int i = 0; i != this->size(); ++i)
```

Comment:
libMesh has different spacing rules. Make this loop look like the others in this file.